src/python/pants/core/goals/test.py

```diff
@@ -321,11 +319,13 @@ async def run_tests(
                 f"{result.address.reference():80}.....{result.test_result.status.value:>10}"
             )
 
-    for result in results:
-        xml_results = result.test_result.xml_results
-        if not xml_results:
-            continue
-        workspace.materialize_directory(DirectoryToMaterialize(xml_results))
```

Comment:
FYI @asherf it's much more performant to not call `.write_digest()` in a for loop. Our bad for not catching this in code review.